File: tags/0.9.8/bcore/command/context_abort.cpp
/*
 * btg Copyright (C) 2005 Michael Wojciechowski.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/*
 * $Id$
 */

#include "context_abort.h"
#include <bcore/util.h>
#include <bcore/helpermacro.h>
#include <iostream>

namespace btg
{
   namespace core
   {

      contextAbortCommand::contextAbortCommand()
         : contextCommand(Command::CN_CABORT, contextCommand::UNDEFINED_CONTEXT, false),
           eraseData_(false)
      {}

      contextAbortCommand::contextAbortCommand(t_int const _context_id, 
                                               bool const _eraseData,
                                               bool const _allContexts)
         : contextCommand(Command::CN_CABORT, _context_id, _allContexts),
           eraseData_(_eraseData)
      {}

      bool contextAbortCommand::serialize(btg::core::externalization::Externalization* _e) const
      {
         BTG_RCHECK( contextCommand::serialize(_e) );

         _e->setParamInfo("flag: erase data", true);
         _e->boolToBytes(this->eraseData_);
         BTG_RCHECK(_e->status());

         return true;
      }

      bool contextAbortCommand::deserialize(btg::core::externalization::Externalization* _e)
      {
         contextCommand::deserialize(_e);
         BTG_RCHECK(_e->status());

         _e->setParamInfo("flag: erase data", true);
         _e->bytesToBool(this->eraseData_);
         BTG_RCHECK(_e->status());

         return true;
      }

      std::string contextAbortCommand::toString() const
      {
         std::string output = contextCommand::toString();

         if (eraseData_)
            {
               output += " Erase data..";
            }
         return output;
      }

      bool contextAbortCommand::eraseData() const
      {
         return eraseData_;
      }

      contextAbortCommand::~contextAbortCommand()
      {}

   } // namespace core
} // namespace btg

File: branches/2.x-branch/UI/www/php/htdocs/bcore/command/context_abort.php
<?php
/*
 * btg Copyright (C) 2005 Michael Wojciechowski.
 * PHP client written by Johan Ström.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

/*
 * $Id: context_abort.php,v 1.1.2.2 2006/06/01 00:38:13 jstrom Exp $
 */
if(!defined("BTG_BCORE_PATH"))
	define("BTG_BCORE_PATH", "bcore");

require_once(BTG_BCORE_PATH."/command/command.php");

class contextAbortCommand extends contextCommand
{
	private $eraseData; // bool
	public function contextAbortCommand($context_id=contextCommand::UNDEFINED_CONTEXT, $eraseData = false, $allContexts = false)
	{
		parent::__construct(Command::CN_CABORT, $context_id, $allContexts);
		$this->eraseData = $eraseData;
	}

	public function serialize(&$a = array())
	{
		$a = parent::serialize();
		$this->boolToBytes($a, $this->eraseData);
		return $a;
	}

	public function deserialize(&$data)
	{
		parent::deserialize($data);
		$this->bytesToBool($this->eraseData, $data);
	}

	public function eraseData()
	{
		return $this->eraseData;
	}
}

?>
